### PR TITLE
fix types

### LINKED ISSUE #36: Correct typings to allow undefined
These functions can return undefined:

https://github.com/microsoft/vscode-windows-process-tree/blob/d15a627301622ea9df9e316be79202f53f0fbcb3/lib/index.ts#L39
https://github.com/microsoft/vscode-windows-process-tree/blob/d15a627301622ea9df9e316be79202f53f0fbcb3/lib/index.ts#L62

### PR DESCRIPTION
**Pull request by meganrogge**
fix #36

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -33,20 +33,34 @@ const MAX_FILTER_DEPTH = 10;
  * @param processList The list of processes
  * @param maxDepth The maximum depth to search
  */
-export function buildProcessTree(rootPid: number, processList: IProcessInfo[], maxDepth: number): IProcessTreeNode {
+export function buildProcessTree(rootPid: number, processList: IProcessInfo[] | undefined, maxDepth: number): IProcessTreeNode | undefined {
```
Comment (meganrogge) [inline lib/index.ts:36]: update dts return types for both functions

Comment (meganrogge) [inline lib/index.ts:36]: update dts return types for both functions

File: lib/index.ts [hunk 2]
```diff
@@ -68,14 +82,22 @@ export function filterProcessList(rootPid: number, processList: IProcessInfo[],
...
+    if (list) {
+      children.push(list);
+    }
+  }
+  return children.reduce((prev, current) => prev.concat(current), [rootProcess]);
 }
 
 function getRawProcessList(
   pid: number,
   queue: RequestQueue,
```
Comment (meganrogge) [inline lib/index.ts:98]: allow RequestQueue's callback to be undefined then can revert the below

Comment (meganrogge) [inline lib/index.ts:98]: allow RequestQueue's callback to be undefined then can revert the below

File: lib/test.ts [hunk 3]
```diff
@@ -99,34 +99,34 @@ describe('getProcessList', () => {
 
   it('should return a list containing this process', (done) => {
     getProcessList(process.pid, (list) => {
-      assert.strictEqual(list.length, 1);
-      assert.strictEqual(list[0].name, 'node.exe');
-      assert.strictEqual(list[0].pid, process.pid);
-      assert.strictEqual(list[0].memory, undefined);
-      assert.strictEqual(list[0].commandLine, undefined);
+      assert.strictEqual(list?.length, 1);
```
Comment (meganrogge) [inline lib/test.ts:102]: ? -> !

Comment (meganrogge) [inline lib/test.ts:102]: ? -> !

**Review by Tyriar — APPROVED** [review]

**Pull request merged**